tests/components/numato/test_numato.py

```diff
@@ -0,0 +1,325 @@
+"""Tests for the numato integration."""
+from copy import copy, deepcopy
+
+from numato_gpio import NumatoGpioError
+import pytest
+
+import homeassistant.components.numato as numato
+from homeassistant.helpers import discovery
+from homeassistant.setup import async_setup_component
+
+from . import numato_mock
+
+from tests.components.switch.common import async_turn_off, async_turn_on
+
+NUMATO_CFG = {
+    "numato": {
+        "discover": ["/ttyACM0", "/ttyACM1"],
+        "devices": [
+            {
+                "id": 0,
+                "binary_sensors": {
+                    "invert_logic": False,
+                    "ports": {
+                        "2": "numato_binary_sensor_mock_port2",
+                        "3": "numato_binary_sensor_mock_port3",
+                        "4": "numato_binary_sensor_mock_port4",
+                    },
+                },
+                "sensors": {
+                    "ports": {
+                        "1": {
+                            "name": "numato_adc_mock_port1",
+                            "source_range": [100, 1023],
+                            "destination_range": [0, 10],
+                            "unit": "mocks",
+                        }
+                    },
+                },
+                "switches": {
+                    "invert_logic": False,
+                    "ports": {
+                        "5": "numato_switch_mock_port5",
+                        "6": "numato_switch_mock_port6",
+                    },
+                },
+            }
+        ],
+    }
+}
+
+
+def mockup_raise(*args, **kwargs):
+    """Mockup to replace regular functions for error injection."""
+    raise NumatoGpioError("Error mockup")
+
+
+def mockup_return(*args, **kwargs):
+    """Mockup to replace regular functions for error injection."""
+    return False
+
+
+@pytest.fixture
+def config():
+    """Provide a copy of the numato domain's test configuration.
+
+    This helps to quickly change certain aspects of the configuration scoped
+    to each individual test.
+    """
+    return deepcopy(NUMATO_CFG)
+
+
+@pytest.fixture
+def numato_fixture():
+    """Inject the numato mockup into numato homeassistant module."""
+    module_mock = copy(numato_mock.NumatoModuleMock)
+    numato.__dict__["gpio"] = module_mock
+    yield module_mock
+    module_mock.cleanup()
+
+
+@pytest.mark.asyncio
```

Comment:
Well, I added it for a reason, noted in the commit comment. Without it, the numato_fixture executed the teardown-code after the yield (cleanup of the devices) too early, namely during test execution. Consequently the test failed, because the device object was already gone.
Any advice how to keep the fixture running without the decorator?

Reply:
The tests run ok for me without the decorator. Check if we need to await `hass.async_block_till_done` done at more places.

Reply:
Okay, you're right. I can't reproduce it myself. Changing it.